generated/nidaqmx/_lib.py

```diff
@@ -46,13 +46,13 @@ def _setter(self, val):
 
 class CtypesByteString:
     """
-    Custom argtype that automatically converts unicode strings to ASCII
+    Custom argtype that automatically converts unicode strings to UTF-8
     strings in Python 3.
     """
     @classmethod
     def from_param(cls, param):
         if isinstance(param, str):
-            param = param.encode('ascii')
+            param = param.encode('utf-8')
```

Comment:
`nicaiu.dll` does not use UTF-8. I am 100% sure this will generate [mojibake](https://en.wikipedia.org/wiki/Mojibake), unless you are using code points that are the same in Latin-1 and UTF-8.

Please see [nidaqmx_internal/errors.py](https://codesearch.natinst.com/search/text?q=nidaqmx_internal%2Ferrors.py&fold_case=auto&regex=true&context=true) and [test_cerrorinfo.py](https://codesearch.natinst.com/search/text?q=test_cerrorinfo.py&fold_case=auto&regex=true&context=true). Try changing `errors.py` to use UTF-8 and then run pytest on `test_cerrorinfo.py`. It won't work.

It also doesn't work for Chinese, Japanese, or Korean without changing the system locale.

Reply:
@bkeryan I have updated to load `nicai_utf8.dll`. Do we want to add a simulated device with non ASCII device name so that it can be covered in some of the regression tests?

Reply:
> I have updated to load nicai_utf8.dll.

I think this is the right long-term solution, but in the short term we probably need to do one of the following:
- Fall back to MBCS if the UTF-8 DLL is not available
- Always use MBCS

Note that we can still enable some internationalization with MBCS, within the limits of the selected system locale. I'm not 100% sure the best way to query the encoding, but `locale.getlocale()[1]` looks like a good starting point. The other methods are either problematic (`locale.getpreferredencoding()`) or require a recent Python.

You could add an encoding property to `DaqLibImporter`. If the UTF-8 DLL is available, set the encoding property to `"utf-8"`, otherwise set it to `locale.getlocale()[1]`.

> Do we want to add a simulated device with non ASCII device name so that it can be covered in some of the regression tests?

This would make sense, but I would prefer for this to be in a separate PR once we have confirmed that the DLL change doesn't break the internal test suite.

We also need to update the internal tests that I just added.

Reply:
More thoughts on this:

If we support both DLLs (via automatic fallback), we should test with both DLLs. That means we need some sort of feature toggle to override the automatic fallback.

If we support only UTF-8 on Windows and MBCS on Linux, we don't need to test with both DLLs on either OS. However, if there are Windows systems with only the MBCS DLL installed, this would cause a regression on those systems. I don't know if there are systems like this. The UTF-8 DLL was used for LV NXG, so it's not particularly new.

A feature toggle could use registry on Windows and INI on Linux like our driver feature toggles do, or it could potentially use something simpler like environment variables or a `.env` file. The MeasurementLink Python module is using `.env` for feature toggles via the `decouple` module.

I don't know whether NI drivers officially support internationalization or localization on Linux. If it's not supported, it would be better to keep using ASCII on Linux and only support internationalization on Windows.

Modern OSes have better support for UTF-8 locales than when NI introduced the UTF-8 version of this DLL. Win10 has opt-in support for UTF-8 ANSI code page but this is not enabled by default. Most Linux distros default to locales with UTF-8 encodings. I don't know if the MBCS DLL supports these configurations. Even if the string conversions work with UTF-8 in this case, localized strings for errors and such may be in the wrong encoding.

Reply:
> The UTF-8 DLL was used for LV NXG, so it's not particularly new.

Seeing the comment on `TaskHandle` about NI-DAQmx 8.9 made me wonder, do we still support any DAQmx versions that don't have `nicai_utf8.dll`? I looked at the history in version control and it was added in DAQmx 16.4, which is newer than I thought. This change as it currently is would break compatibility with DAQmx <16.4.

Reply:
@bkeryan meaning we should let it load `nicai_utf8.dll` and if causes an error (let's say having DAQmx <16.4), then we'll fallback to `nicaiu.dll`? and this only applies to Windows since we're not supporting UTF-8 in Linux version.

Reply:
Yes. That would preserve compatibility with DAQmx versions from 2016 and older.

Also, if we support both DLLs, we should test both DLLs. It's easier to test the fallback case if there is some sort of override.